plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx

```diff
@@ -99,6 +101,26 @@ const useStyles = makeStyles(theme => ({
   },
 }));
 
+const HeaderTitle = () => {
+  const showBackButton = document.referrer;
```

Comment:
I'm not sure I like this concept, of relying on the _actual_ back functionality. Browsers do that just fine already.

It's a bit iffy to use referrer isn't it, what if you open in a new tab or it's in an iframe or whatnot. I'm thinking, if we want to have this type of almost cross-page "sibling navigation", then maybe we should have a strategy for how to do it in a concerted way.

Google's cloud console has a similar thing with back buttons in the title bar and I agree that the concept is nice. But what should "power" the presence and target of the arrow button? It's not so much a "back", as it's an "up". Should it be a global thing for ALL `<Header>` that they look for a certain globally agreed-upon query string for example? What if the originating page linked to `/other-page?nav-previous=/my/own/current/path` or something like that? Then you could send that link to somebody else and they'd view that graph "in the right context" so to speak.

Suggestions welcome. But in the current shape of the PR, I have some doubts. 🤔

Reply:
Hi @freben, first of all thank you for the review.

> what if you open in a new tab or it's in an iframe or whatnot
in that case, the back button is not displayed. That's why I use `showBackButton` as a boolean to render the icon button.

> It's not so much a "back"
At practive, I think is just a "back". If you try to navigate [here](https://console.cloud.google.com/marketplace/product/elastic-prod/elastic-cloud) using a fresh tab, you will see that clicking on the back arrow will bring you to the "new tab" section of your browser, instead of another section of the GCP console.
IMO this behaviour is bad, and I avoided it using the referrer.

> Suggestions welcome.
IMO each time that the back button should appear, we could define there you want to be redirect once you click the back button...but I don't think is the right solution: in this case I think there are multiple ways to reach the same page.